### PR TITLE
Add auto generated documents with fsdocs.

### PR DESCRIPTION
**Pull request by joemphilips**
Based on https://github.com/joemphilips/DotNetLightning/pull/191

### CODE REVIEW

**Comment by knocte** [comment]
> stop supporting netstandard2.0
> Since we don't have an intention to support other env such as .NET Frameworks,

What? geewallet can be compiled with Mono.

**Comment by joemphilips** [comment]
`netstandard2.1` is enough to support Mono, Xamarin, .NET Core 3.x.
So I suppose there is no problem for geewallet.

https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-1-0#net-standard-not-deprecated
(Sorry that commit msg is not easy to understand.)

**Comment by knocte** [comment]
.netstandard2.1 is not supported by Mono version 6.8 (which is the one you get in ubuntu), or xbuild

**Comment by joemphilips** [comment]
> .netstandard2.1 is not supported by Mono version 6.8 (which is the one you get in ubuntu), or xbuild

Alright than I suppose I will target only `netstandard2.0` instead ...

**Comment by knocte** [comment]
> Alright than I suppose I will target only netstandard2.0 instead ...

Sounds good. I don't see any reason to support .netstandard2.1.

**Comment by knocte** [comment]
That logo is badass! did you design it?

**Comment by joemphilips** [comment]
> That logo is badass! did you design it?

Done in 5 minutes 😎

**Comment by joemphilips** [comment]
This PR is getting huge. I think it is now good enough for starting point.
@knocte would you mind reviewing? Otherwise I will merge in couple of days.

**Comment by joemphilips** [comment]
Rendered version can be found at
https://joemphilips.github.io/DotNetLightning/

File: src/DotNetLightning.Core/DotNetLightning.Core.fsproj [hunk 1]
```diff
@@ -107,8 +110,7 @@
   <!-- this is a workaround only needed for nuget push (so, not Mono, but just "dotnet nuget"), for more info see
        https://github.com/joemphilips/DotNetLightning/issues/14 and https://github.com/joemphilips/DotNetLightning/commit/c98307465f647257df1438beadb4cabc7db757f2
        and https://github.com/NuGet/Home/issues/3891 and https://github.com/NuGet/Home/issues/3891#issuecomment-377319939 -->
-  <Target Name="CopyProjectReferencesToPackage" DependsOnTargets="ResolveReferences"
-          Condition="'$(MSBuildRuntimeType)'!='Mono'">
+  <Target Name="CopyProjectReferencesToPackage" DependsOnTargets="ResolveReferences" Condition="'$(MSBuildRuntimeType)'!='Mono'">
```
Comment (knocte) [inline src/DotNetLightning.Core/DotNetLightning.Core.fsproj:113]: for git-blame cleanliness, this hunk should be dropped

File: src/DotNetLightning.Core/Serialization/Features.fs [hunk 2]
```diff
@@ -205,10 +209,11 @@ module internal Feature =
         }
         |> Set
 
-
+/// Feature bits described in [BOLT 9](https://github.com/lightning/bolts/blob/master/09-features.md)
+/// Use `HasFeature` to see if it has specific feature or not.
 [<StructuredFormatDisplay("{PrettyPrint}")>]
 type FeatureBits private (bitArray: BitArray) =
-    member __.BitArray
+    member _.BitArray
```
Comment (knocte) [inline src/DotNetLightning.Core/Serialization/Features.fs:216]: this line above doesn't need to change either

**Comment by knocte** [comment]
LGTM except 2 nits pointed out above

File: src/DotNetLightning.Core/Utils/Primitives.fs [hunk 3]
```diff
@@ -567,20 +596,37 @@ module Primitives =
...
+    [<Obsolete>]
     type UserId = UserId of uint64
+
+    [<Obsolete>]
     type Delimiter = Delimiter of array<byte>
 
+    /// Each lightning node can specify its own RGB color in node_announcement.
+    /// Used for e.g. lightning node explorer.
+    /// <seealso cref="DotNetLightning.Serialization.Msgs.UnsignedNodeAnnouncementMsg" />
+    [<Struct>]
```
Comment (knocte) [inline src/DotNetLightning.Core/Utils/Primitives.fs:608]: this Struct addition is unrelated?

Comment (joemphilips) [reply src/DotNetLightning.Core/Utils/Primitives.fs:608]: I have split into independent commit.
(Please don't say we should create another PR. independent commit is enough.)
